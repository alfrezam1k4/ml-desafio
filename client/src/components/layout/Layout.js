import React from 'react';
import { Route, Switch, withRouter } from 'react-router-dom';
import AppBar from './AppBar';
import Background from './Background';
import { ItemDetails, ItemsResult } from '../Items';
import './Layout.scss';

const Layout = () => {
  return (
    <Background>
      <AppBar />
      <main id="main-content" className="main-layout" role="main">
        <div className="content-layout">
          <Switch>
            <Route exact path="/items/:id" component={ItemDetails} />
            <Route exact path="/items" component={ItemsResult} />
          </Switch>
        </div>
      </main>
    </Background>
  );
};

export default withRouter(Layout);
