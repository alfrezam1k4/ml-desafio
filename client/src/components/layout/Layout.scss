@import '../../utils/variables.scss';

.main-layout {
  display: flex;
  flex: auto;
  flex-direction: column;
  height: 100%;
}
.content-layout {
  display: flex;
  flex-direction: column;
  margin: 0 auto;
  width: calc(100% - 80px);
  max-width: 1120px;
}
.background {
  min-height: 100vh;
  background-color: $lighten_gray;
}
