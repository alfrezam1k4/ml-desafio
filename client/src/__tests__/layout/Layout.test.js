import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Layout from '../../components/layout/Layout';

describe('<Layout />', () => {
  it('renders correctly', () => {
    const component = renderer.create(
      <MemoryRouter>
        <Layout />
      </MemoryRouter>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
