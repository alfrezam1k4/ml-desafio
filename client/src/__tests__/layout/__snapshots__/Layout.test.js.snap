// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout /> renders correctly 1`] = `
<div
  className="background"
>
  <header
    className="app-bar"
  >
    <div
      className="app-bar-content"
    >
      <form
        action="/items"
        className="search-container"
        method="get"
      >
        <a
          href="/"
          onClick={[Function]}
        >
          <img
            alt="Logo Mercado Libre"
            className="header-logo"
            src="Logo_ML@2x.png.png"
          />
        </a>
        <input
          className="search-input"
          name="search"
          placeholder="Nunca dejes de buscar"
        />
        <button
          className="search-button"
        >
          <img
            alt="Buscar"
            src="ic_Search@2x.png.png"
          />
        </button>
      </form>
    </div>
  </header>
  <main
    className="main-layout"
    id="main-content"
    role="main"
  >
    <div
      className="content-layout"
    />
  </main>
</div>
`;
